lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -791,21 +946,22 @@ def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(
         dict(
-            access_logs_enabled=dict(required=False, type='bool'),
-            access_logs_s3_bucket=dict(required=False, type='str'),
-            access_logs_s3_prefix=dict(required=False, type='str'),
-            deletion_protection=dict(required=False, default=False, type='bool'),
-            idle_timeout=dict(required=False, type='int'),
-            listeners=dict(required=False, type='list'),
+            access_logs_enabled=dict(type='bool'),
+            access_logs_s3_bucket=dict(type='str'),
+            access_logs_s3_prefix=dict(type='str'),
+            deletion_protection=dict(default=False, type='bool'),
+            idle_timeout=dict(type='int'),
+            listeners=dict(type='list'),
```

Comment:
I'm still convinced that my suggested fix commented on line 727 resolves this. Here is my reproducer, consistently fails:
```
---
- hosts: localhost
  connection: local
  vars:
    elb_app_lb_name: "{{ elb_app_lb_name }}"
    tg_name: "{{ tg_name }}"
  tasks:
    - name: create an application lb
      elb_application_lb:
        state: present
        name: "{{ elb_app_lb_name }}"
        subnets:
          - subnet-17bbda4c
          - subnet-5c7ecc60
        security_groups: sg-29d13055
        listeners:
          - Protocol: HTTP
            Port: 80
            DefaultActions:
              - 'Type': 'forward'
                'TargetGroupName': '{{ tg_name }}'
    - name: update subnets
      elb_application_lb:
        state: present
        name: "{{ elb_app_lb_name }}"
        subnets:
          - subnet-cbdf56ae
          - subnet-17bbda4c
          - subnet-5c7ecc60
        security_groups: sg-29d13055
```
Traceback:
```
TASK [update subnets] **************************************************************************************************
task path: /Users/shertel/Workspace/ansible/my_playbooks/elb_application_lb/iterate_over_NoneType.yml:22
Using module file /Users/shertel/Workspace/ansible/lib/ansible/modules/cloud/amazon/elb_application_lb.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: shertel
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/shertel/.ansible/tmp/ansible-tmp-1499709521.38-260285283136193 `" && echo ansible-tmp-1499709521.38-260285283136193="` echo /Users/shertel/.ansible/tmp/ansible-tmp-1499709521.38-260285283136193 `" ) && sleep 0'
<127.0.0.1> PUT /var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/tmpiiu8JI TO /Users/shertel/.ansible/tmp/ansible-tmp-1499709521.38-260285283136193/elb_application_lb.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /Users/shertel/.ansible/tmp/ansible-tmp-1499709521.38-260285283136193/ /Users/shertel/.ansible/tmp/ansible-tmp-1499709521.38-260285283136193/elb_application_lb.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/Users/shertel/Workspace/ansible/venv/python2.7/bin/python /Users/shertel/.ansible/tmp/ansible-tmp-1499709521.38-260285283136193/elb_application_lb.py; rm -rf "/Users/shertel/.ansible/tmp/ansible-tmp-1499709521.38-260285283136193/" > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):


  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py", line 1005, in <module>
    main()
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py", line 1000, in main
    create_or_update_elb(connection, connection_ec2, module)
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py", line 895, in create_or_update_elb
    listener_changed = create_or_update_elb_listeners(connection, module, elb)
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py", line 704, in create_or_update_elb_listeners
    listeners = ensure_listeners_default_action_has_arn(connection, module, module.params.get("listeners"))
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py", line 487, in ensure_listeners_default_action_has_arn
    for listener in listeners:
TypeError: 'NoneType' object is not iterable

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py\", line 1005, in <module>\n    main()\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py\", line 1000, in main\n    create_or_update_elb(connection, connection_ec2, module)\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py\", line 895, in create_or_update_elb\n    listener_changed = create_or_update_elb_listeners(connection, module, elb)\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py\", line 704, in create_or_update_elb_listeners\n    listeners = ensure_listeners_default_action_has_arn(connection, module, module.params.get(\"listeners\"))\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_CNtIaB/ansible_module_elb_application_lb.py\", line 487, in ensure_listeners_default_action_has_arn\n    for listener in listeners:\nTypeError: 'NoneType' object is not iterable\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 0
}

PLAY RECAP *************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=1
```